File: napakalaki(ruby)/pkg/napakalaki-0.0.1/lib/PruebaNapakalaki.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.


#monster = Array.new
=begin
require 'prize'
require 'bad_consequence'
require 'treasure_kind'
require 'monster'


# Métodos de consultas-----------------------------------------------------------------------------
# Consulta1
#def consulta1(monster)
  #aux = Array.new
  #monster.each do |m|
    #if (m.combatLevel>10) then
    #aux << m
    #end
  #end
  #return aux
#end

#Consulta2
#def consulta2(monster)
  #aux = Array.new
  #monster.each do |m|
    #if (m.bc.levels>0) then
    #aux << m
    #end
  #end
  #return aux
#end

#Consulta3
def consulta3(monster)
  aux = Array.new
  monster.each do |m|
    if (m.price.levels>1) then
    aux << m
    end
  end
  return aux
end

#Consulta4
def consulta4(monster)
  aux = Array.new
  monster.each do |m|
    if (!m.bc.specificVisibleTreasure.empty? || !m.bc.specificHiddenTreasure.empty?) then
    aux << m
    end
  end
  return aux
end

#puts "Los mostrous con nivel de combate superior a 10: "
#puts consulta1(monster)

#puts "Los mostrous con mal rollo que implican solo  perdidas de niveles: "
#puts consulta2(monster)

#puts "Los mostrous con un buen rollo que implican una ganancia de niveles superior a 1: "
#puts consulta3(monster)

#puts "Los mostrous con un mal rollo que implican perdida de un determinado tipo de tesoros ya sea visibles y/o ocultos: "
#puts consulta4(monster)

=end

File: napakalaki/src/Napakalaki/PruebaNapakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Napakalaki;

import java.util.ArrayList;
import java.util.Arrays;

/**
 *
 * @author marcos
 */
public class PruebaNapakalaki {

    /**
     * @param args the command line arguments
     */
    
    
    /*//NIVEL DE COMBATE SUPERIOR A 10 = consulta1
   static ArrayList <Monster> consulta1 (ArrayList<Monster> monstruos){
        ArrayList<Monster> aux= new ArrayList();
        for(Monster m: monstruos){
             if( m.getCombatLevel()>10)
             {
                aux.add(m); 
             }
         }
        return aux;
    }
    
       //MAL ROLLO CON PÉRDIDA DE NIVELES SUPERIOR A 1 = consulta2
   static ArrayList <Monster> consulta2 (ArrayList<Monster> monstruos){
        ArrayList<Monster> aux= new ArrayList();
        for(Monster m: monstruos){
             if( m.getBadConsequence().getLevels()>0)
             {
                aux.add(m); 
             }
         }
        return aux;
    }
   
          //BUEN ROLLO CON GANANCIAS DE NIVELES SUPERIOR A 1 = consulta3
   static ArrayList <Monster> consulta3 (ArrayList<Monster> monstruos){
        ArrayList<Monster> aux= new ArrayList();
        for(Monster m: monstruos){
             if( m.getPrize().getLevel()>1)
             {
                aux.add(m); 
             }
         }
        return aux;
    }
   
   //MAL ROLLO CON PERDIDA DE UN TIPO DE TESORO VISIBLE Y/O OCULTOS = consulta4
   static ArrayList <Monster> consulta4 (ArrayList<Monster> monstruos){
        ArrayList<Monster> aux= new ArrayList();
        for(Monster m: monstruos){
             if( !m.getBadConsequence().getTesorosEspecificosVisible().isEmpty() || !m.getBadConsequence().getTesorosEspecificosOcultos().isEmpty() ){
                  aux.add(m);
             }
         }
        return aux;
    }*/
   
    /*public static void main(String[] args) {
        
        ArrayList<Monster> monstruos = new ArrayList();
        ArrayList<Life> life = new ArrayList();
        
        //monstruo 1
        BadConsequence bad = new BadConsequence("Pierdes tu armadura visible y otra oculta",0,new ArrayList(Arrays.asList(TreasureKind.ARMOR)),new ArrayList(Arrays.asList(TreasureKind.ARMOR))); 
        Prize prize = new Prize(2,1); 
        monstruos.add(new Monster("3 Byakhees de bonanza", 8, bad, prize));

        
        
         
         System.out.print("\n Los monstrous mayores de nivel 10: "+consulta1(monstruos));
         System.out.print("\n Los monstrous con pérdidas de niveles superior a 1: "+consulta2(monstruos));
         System.out.print("\n Los monstrous con ganancias de niveles superior a 1: "+consulta3(monstruos));
         System.out.print("\n Los monstrous con pérdida de un tipo de tesoro visible y/o oculto: "+consulta4(monstruos));
        
        
        //EXAMEN
        ArrayList<Player> jugadores = new ArrayList();
        jugadores.add(new Player("Marcos"));
        jugadores.add(new Player("Marcos2"));
        //life.add(new Life());
        //life.add(new Life());
        //jugadores.get(0).addFriend(monstruos.get(0));
        //jugadores.get(1).addFriend(monstruos.get(1));
        //jugadores.get(0).addLife(life.get(0));
        //jugadores.get(1).addLife(life.get(1));
        System.out.print("El jugador 1 con nivel de combate:"+jugadores.get(0).getCombatLevel());
        System.out.print("El jugador 1 con nivel de combate:"+jugadores.get(1).getCombatLevel());
        //EXAMEN
    }*/
}
